File: psp/psp1Eva/PSP1Eva/ServidorSpring/src/main/java/com/example/servidorspring/spring/errors/ControlErrores.java
package com.example.servidorspring.spring.errors;


import com.example.servidorspring.domain.exceptions.BaseDatosCaidaException;
import com.example.servidorspring.domain.exceptions.NotFoundException;
import domain.errors.ApiError;
import jakarta.servlet.ServletException;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.dao.DataIntegrityViolationException;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Component;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;

import java.io.IOException;
import java.nio.file.AccessDeniedException;
import java.time.LocalDateTime;

@ControllerAdvice
@Component("MisErrores")
public class ControlErrores extends ResponseEntityExceptionHandler  {


    @ResponseStatus(HttpStatus.NOT_FOUND)
    @ExceptionHandler(NotFoundException.class)
    public ResponseEntity<ApiError> handleNotFoundException(NotFoundException e) {
        return ResponseEntity.status(HttpStatus.NOT_FOUND)
                .body(new ApiError(e.getMessage(), LocalDateTime.now()));
    }


    @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
    @ExceptionHandler(BaseDatosCaidaException.class)
    public ResponseEntity<ApiError> handleBaseDatosCaidaException(BaseDatosCaidaException e) {
        return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
                .body(new ApiError(e.getMessage(), LocalDateTime.now()));
    }
}

File: servidorExamenMoviles2EVA/examen/examen/src/main/kotlin/com/example/examen/errores/ControlErrores.kt
package com.example.examen.errores

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Component
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.ResponseStatus

@ControllerAdvice
@Component("MisErrores")
class ControlErrores {


    @ResponseStatus(HttpStatus.NOT_FOUND)
    @ExceptionHandler(
        NotFoundException::class
    )
    fun handleNotFoundException(e: NotFoundException): ResponseEntity<ApiError?>? {
        return ResponseEntity.status(HttpStatus.NOT_FOUND)
            .body<ApiError?>(ApiError(e.message, "debug", "404"))
    }

}
